sklearn/linear_model/cd_fast.pyx

```diff
@@ -201,26 +233,26 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 
                 if w_ii != 0.0:
                     # R += w_ii * X[:,ii]
-                    daxpy(n_samples, w_ii,
-                          <DOUBLE*>(X.data + ii * n_samples * sizeof(DOUBLE)),
-                          1, <DOUBLE*>R.data, 1)
+                    axpy(n_samples, w_ii,
+                        <floating*>(X.data + ii * n_samples * sizeof(floating)),
```

Comment:
Could you try defining `X_data` and `y_data` above and then just using `&X_data[ii * n_samples]`??